Libraries/Renderer/src/renderers/native/NativeMethodsMixin.js

```diff
@@ -139,6 +139,12 @@ var NativeMethodsMixin = {
    * Manipulation](docs/direct-manipulation.html)).
    */
   setNativeProps: function(nativeProps: Object) {
+    if (!this.viewConfig) {
+      var ctor = this.constructor;
+      var componentName = ctor.displayName || ctor.name || '<Unknown Component>';
+      throw new Error(componentName + ' "viewConfig" is not defined.');
```

Comment:
Can you use `invariant` here? Because you already have checked for the error, please use `invariant(false, ...`